src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -13,6 +12,11 @@
     public BatchItem(final JSONObject event) {
         this.response = new BatchItemResponse();
         this.event = event;
+
+        Optional.
+                of(event.optJSONObject("metadata"))
+                .map(e -> e.optString("eid"))
```

Comment:
But it will not be null. `map` method will treat `null` as empty and code inside `ifPresent` will not be executed. And `eid` will stay with it's default value (empty string) :)